src/main/java/com/google/sps/servlets/Util.java

```diff
@@ -92,8 +92,10 @@ public static String getAPIKey() throws IOException {
   private static String getRefreshToken(String userId) throws IOException, TooManyResultsException {
     Query query = new Query("RefreshToken").setFilter(FilterOperator.EQUAL.of("userId", userId));
     PreparedQuery results = datastore.prepare(query);
-    System.out.println(results.countEntities());
     Entity entity = results.asSingleEntity();
+    if (entity == null) {
+        throw new IOException("could not get refresh token");
```

Comment:
I think "no refresh token found" is one of the expected state in our auth flow. So instead of throwing an exception, it would be better to let the function return ```Optional<String>```, so if the refresh token is not there, it will return ```Optional.empty()```.

Reply:
+1

Reply:
will change to return an optional object -- it is throwing an exception here because some servlets recognize an ioexception being thrown as a sign to redirect to the authorize page....